include/metal/list/partition.hpp

```diff
@@ -35,9 +43,71 @@ namespace metal {
     ///
     /// ### See Also
     /// \see list, copy_if, remove_if
+#if !defined(METAL_WORKAROUND)
+    template<class seq, class lbd>
+    using partition = typename detail::_partition<lbd>::template type<seq>;
+#else
     template<class seq, class lbd>
     using partition =
         metal::pair<metal::copy_if<seq, lbd>, metal::remove_if<seq, lbd>>;
+#endif
```

Comment:
This will end up in a big mess because it implies different dependency graphs for different compilers.